cmd/antrea-agent/agent.go

```diff
@@ -999,6 +1004,10 @@ func run(o *Options) error {
 		go nodeLatencyMonitor.Run(stopCh)
 	}
 
+	if linkDetector != nil {
+		go linkDetector.Run(stopCh)
+	}
```

Comment:
could we start the link detector before the controllers that consume it (Egress / ServiceExternalIP), that would make more sense to me?

Reply:
the motivation is to share the detector instance between ServiceExternalIP and Egress controller. Otherwise we may need to create separate instances for the two controllers.

Reply:
I was just suggesting calling `go linkDetector.Run(stopCh)` before we call `go externalIPController.Run(stopCh)` and `go egressController.Run(stopCh)`. It should not impact the ability to share the instance across both controllers?